inst/include/cpp11/logicals.hpp

```diff
@@ -29,22 +29,23 @@ inline SEXP r_vector<r_bool>::valid_type(SEXP data) {
 
 template <>
 inline r_bool r_vector<r_bool>::operator[](const R_xlen_t pos) const {
-  return is_altrep_ ? static_cast<r_bool>(LOGICAL_ELT(data_, pos)) : data_p_[pos];
+  return is_altrep_ ? LOGICAL_ELT(data_, pos) : data_p_[pos];
```

Comment:
`LOGICAL_ELT()` and `data_p_[]` now return the same type (an `int`). So in theory the right thing to do would be to wrap the whole ternary condition in `static_cast<r_bool>`, but I'm relying on the implicit cast to `r_bool` instead - this seems to be what we do elsewhere.

I am not sure what best practice is here, rely on implicit casts to the return value, or be explicit with `static_cast<r_bool>`? I think I'd rather see the explicit cast, so there is no guessing involved. I'll probably do a follow up PR to try this out everywhere.

Reply:
Apparently this is a Core Guideline to avoid implicit conversions where possible http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c164-avoid-implicit-conversion-operators

Reply:
I agree that explicit is better.